### PR TITLE
Add Subsquid Indexer Pages under Ethereum and Substrate API

### PR DESCRIPTION
**Pull request by themacexpert**
### Description

These are quick start pages under the Ethereum-API and Substrate-API pages to get users quickly up to speed on indexing data on their ContainerChain. For a more comprehensive tutorial, users should see the Subsquid tutorial under the tutorials section. 

### Checklist

- [x] If this page requires a disclaimer, I have added one
- [ ] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/interact/ethereum-api/indexers/index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: Indexers
+description: Learn how to use Indexers to ingest, process, and query EVM data on your Tanssi EVM ContainerChain.
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/index.md:3]: This description is too short. Should be a minimum of 120 characters, but no more than 160

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+---
+title: Index your Tanssi EVM ContainerChain with Subsquid
+description: Learn how to use Subsquid, a query node framework for Substrate-based chains, to index and process EVM data your Tanssi EVM ContainerChain.
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:3]:
```suggestion
description: Learn how to use Subsquid, a query node framework for Substrate-based chains, to index and process EVM data on your Tanssi EVM ContainerChain.
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
+---
+title: Index your Tanssi EVM ContainerChain with Subsquid
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:2]: This title is four characters too long. The title should be no more than 60 characters, but you have to keep in mind that `| Tanssi Docs` is appended to the title so in this case this title really would be `Index your Tanssi EVM ContainerChain with Subsquid | Tanssi Docs`

Also `Your` should be capitalized

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+description: Learn how to use Subsquid, a query node framework for Substrate-based chains, to index and process EVM data your Tanssi EVM ContainerChain.
+---
+
+# Indexing your EVM ContainerChain with Subsquid
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:6]:
```suggestion
# Indexing Your EVM ContainerChain with Subsquid
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+## Checking Prerequisites {: #checking-prerequisites }
+
+For the examples in this guide, you will need to have the following:
+
+ - [Node.js](https://nodejs.org/en/download/){target=_blank} version 16 or newer
+  - [Docker](https://docs.docker.com/get-docker/){target=_blank}
+  - [Squid CLI](https://docs.subsquid.io/squid-cli/installation/){target=_blank}
+
+## Index EVM Data on your ContainerChain {: #index-evm-data-on-your-containerchain}
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:22]:
```suggestion
## Index EVM Data on Your ContainerChain {: #index-evm-data-on-your-containerchain}
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+3. Modify the `src/processor.ts` file to set the data source to the RPC URL of your ContainerChain. Remove the `archive: lookupArchive('eth-mainnet')` line as a Squid archive will not be used. Here, the RPC URL of the Demo EVM ContainerChain is specified. The Squid project will use the RPC endpoint to ingest the relevant data. 
+
+    ```ts
+      const processor = new EvmBatchProcessor();
+      processor.setDataSource({
+        chain: 'https://fraa-dancebox-3001-rpc.a.dancebox.tanssi.network',
+      })
+    ```
+
+
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:60]:
```suggestion
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+This quick-start guide will show you how to create a Subsquid project and configure it to index data on your Tanssi EVM ContainerChain. For a more comprehensive end-to-end tutorial, be sure to check out [Using Subsquid to Index Your ContainerChain](/builders/tutorials/subsquid/){target=_blank}.
+
+## Checking Prerequisites {: #checking-prerequisites }
+
+For the examples in this guide, you will need to have the following:
+
+ - [Node.js](https://nodejs.org/en/download/){target=_blank} version 16 or newer
+  - [Docker](https://docs.docker.com/get-docker/){target=_blank}
+  - [Squid CLI](https://docs.subsquid.io/squid-cli/installation/){target=_blank}
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:20]:
```suggestion
 - [Node.js](https://nodejs.org/en/download/){target=_blank} version 16 or newer
 - [Docker](https://docs.docker.com/get-docker/){target=_blank}
 - [Squid CLI](https://docs.subsquid.io/squid-cli/installation/){target=_blank}
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+      - [Quickstart: EVM chains](https://docs.subsquid.io/quickstart/quickstart-ethereum/){target=_blank}
+      - [Quickstart: generate from ABI](https://docs.subsquid.io/quickstart/quickstart-abi/){target=_blank}
+
+2. Navigate into the root directory of your Squid project and install dependencies by running: 
+
+    ```bash
+    npm ci
+    ```
+
+3. Modify the `src/processor.ts` file to set the data source to the RPC URL of your ContainerChain. Remove the `archive: lookupArchive('eth-mainnet')` line as a Squid archive will not be used. Here, the RPC URL of the Demo EVM ContainerChain is specified. The Squid project will use the RPC endpoint to ingest the relevant data.
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:51]:
```suggestion
3. Modify the `src/processor.ts` file to set the data source to the RPC URL of your ContainerChain. Remove the `archive: lookupArchive('eth-mainnet')` line as a Squid archive will not be used. Here, the RPC URL of the demo EVM ContainerChain is specified. The Squid project will use the RPC endpoint to ingest the relevant data
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+    ```ts
+      const processor = new EvmBatchProcessor();
+      processor.setDataSource({
+        chain: 'https://fraa-dancebox-3001-rpc.a.dancebox.tanssi.network',
+      })
+    ```
+
+
+4. Launch Postgres and detach by running:
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:61]:
```suggestion
4. Launch Postgres by running:
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    sqd process
+    ```
+
+6. Open a separate terminal window in the same directory, then start the GraphQL server: 
+
+    ```bash
+    sqd serve
+    ```
+
+7. You can now run queries, such as the sample query below, against your Squid on the GraphQL playground at [http://localhost:4350/graphql](http://localhost:4350/graphql){target=_blank}. If you've modified the template EVM squid to index different data, you'll need to modify this query accordingly.
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:79]:
```suggestion
7. You can now run queries, such as the sample query below, against your Squid on the GraphQL playground at [http://localhost:4350/graphql](http://localhost:4350/graphql){target=_blank}. If you've modified the template EVM squid to index different data, you'll need to modify this query accordingly
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+        block
+        id
+        txHash
+        value
+      }
+    }
+    ```
+
+
+And that's all you have to do to configure your Subsquid project to index EVM data on your Tanssi EVM ContainerChain! Now you can update the `schema.graphql`, `main.ts`, and `src/processor.ts` files to index the data you need for your project!
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:94]:
```suggestion
And that's all you have to do to configure your Subsquid project to index EVM data on your Tanssi EVM ContainerChain! Now you can update the `schema.graphql`, `src/main.ts`, and `src/processor.ts` files to index the data you need for your project!
```

File: builders/interact/ethereum-api/indexers/subsquid.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+If you're interested in a more comprehensive step-by-step tutorial to get started indexing data your ContainerChain, you can check out the [Using Subsquid to Index Your ContainerChain](/builders/tutorials/subsquid/){target=_blank} tutorial!
```
Comment (eshaben) [inline builders/interact/ethereum-api/indexers/subsquid.md:96]:
```suggestion
If you're interested in a more comprehensive step-by-step tutorial to get started indexing data for your ContainerChain, you can check out the [Using Subsquid to Index Your ContainerChain](/builders/tutorials/subsquid/){target=_blank} tutorial!
```

File: builders/interact/substrate-api/indexers/index.md [hunk 13]
```diff
@@ -0,0 +1,8 @@
+---
+title: Indexers
+description: Learn how to use Indexers to ingest, process, and query Substrate data on your Tanssi Substrate ContainerChain.
```
Comment (eshaben) [inline builders/interact/substrate-api/indexers/index.md:3]: This one is also a tad bit too short

File: builders/interact/substrate-api/indexers/index.md [hunk 14]
```diff
@@ -0,0 +1,8 @@
+---
+title: Indexers
```
Comment (eshaben) [inline builders/interact/substrate-api/indexers/index.md:2]: Ideally, this should be different than the other title on the EVM side of things

File: builders/interact/substrate-api/indexers/subsquid.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
+---
+title: Index your Tanssi Substrate ContainerChain with Subsquid
```
Comment (eshaben) [inline builders/interact/substrate-api/indexers/subsquid.md:2]: This is also too long

File: builders/interact/substrate-api/indexers/subsquid.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    ```
+
+7. You can query your template Substrate squid with the below sample query. If you've modified the template Substrate squid to index different data, you'll need to modify this query accordingly.
+
+    ```graphql
+    query MyQuery {
+        accountsConnection(orderBy: id_ASC) {
+          totalCount
+        }
+      }
```
Comment (eshaben) [inline builders/interact/substrate-api/indexers/subsquid.md:75]:
```suggestion
    query MyQuery {
      accountsConnection(orderBy: id_ASC) {
        totalCount
      }
    }
```

**Review by eshaben — CHANGES_REQUESTED** [review]
Left some comments on some minor things. Mostly left comments on the EVM guide, but they all will need to be applied to the Substrate guide too, please!

A couple of things to watch for: there should be no periods at the end of list items and "Your" in titles should be capitalized

I think we need a different icon for the Indexers icon. Sorry, I'm a real stickler about these haha. I can take a look to see if I can find something. Will let you know 🙂

File: builders/interact/substrate-api/indexers/subsquid.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+
+    ```graphql
+    query MyQuery {
+      accountsConnection(orderBy: id_ASC) {
+        totalCount
+      }
+    }
+    ```
+
+And that's all you have to do to configure your Subsquid project to index Substrate data on your Tanssi Substrate ContainerChain! Now you can update the `schema.graphql`, `src/main.ts`, `typgen.json`, and `src/processor.ts` files to index the data you need for your project!
```
Comment (eshaben) [inline builders/interact/substrate-api/indexers/subsquid.md:78]:
```suggestion
And that's all you have to do to configure your Subsquid project to index Substrate data on your Tanssi Substrate ContainerChain! Now you can update the `schema.graphql`, `src/main.ts`, `typegen.json`, and `src/processor.ts` files to index the data you need for your project!
```

File: builders/interact/substrate-api/indexers/subsquid.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+    sqd process
+    ```
+
+6. Open a separate terminal window in the same directory, then start the GraphQL server 
+
+    ```bash
+    sqd serve
+    ```
+
+7. You can query your template Substrate squid with the below sample query. If you've modified the template Substrate squid to index different data, you'll need to modify this query accordingly
```
Comment (eshaben) [inline builders/interact/substrate-api/indexers/subsquid.md:68]:
```suggestion
7. You can query your template Substrate Squid with the below sample query. If you've modified the template Substrate squid to index different data, you'll need to modify this query accordingly
```

**Review by eshaben — COMMENTED** [review]
Looks like the substrate-api index page icons still need to be changed, please! Missed a couple minor things earlier. Aside from these little things, it looks good!

**Comment by themacexpert** [comment]
Thank you @eshaben! fixed

**Review by eshaben — APPROVED** [review]
LGTM!

**Pull request merged**